File: frontend/src/models/guarantee/guaranteesByTypeDTO.ts
// guaranteesByTypeDTO.ts

import { BusinessFundGuarantee } from './businessFundGuarantee';
import { MortgageGuarantee } from './mortgageGuarantee';
import { PersonalGuarantee } from './personalGuarantee';
import { RealEstateGuarantee } from './realEstateGuarantee';
import { VehicleGuarantee } from './vehicleGuarantee';

export interface GuaranteesByTypeDTO {
  businessFundGuarantees: BusinessFundGuarantee[];
  mortgageGuarantees: MortgageGuarantee[];
  personalGuarantees: PersonalGuarantee[];
  realEstateGuarantees: RealEstateGuarantee[];
  vehicleGuarantees: VehicleGuarantee[];
}

File: backend/recouvrex-services/src/main/java/com/recouvrex/process/model/guarantee/GuaranteesByTypeDTO.java
package com.recouvrex.process.model.guarantee;

import java.util.List;

import lombok.Data;

@Data
public class GuaranteesByTypeDTO {
    private List<BusinessFundGuarantee> businessFundGuarantees;
    private List<MortgageGuarantee> mortgageGuarantees;
    private List<PersonalGuarantee> personalGuarantees;
    private List<RealEstateGuarantee> realEstateGuarantees;
    private List<VehicleGuarantee> vehicleGuarantees;
}